ludwig/schema/features/text_feature.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+from marshmallow_dataclass import dataclass
+
+from ludwig.constants import TEXT
+from ludwig.decoders.registry import get_decoder_classes
+from ludwig.encoders.registry import get_encoder_classes
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.preprocessing import BasePreprocessingConfig, PreprocessingDataclassField
+
+
+@dataclass
+class TextInputFeatureConfig(schema_utils.BaseMarshmallowConfig):
+    """TextInputFeatureConfig is a dataclass that configures the parameters used for a text input feature."""
+
+    preprocessing: BasePreprocessingConfig = PreprocessingDataclassField(feature_type=TEXT)
+
+    encoder: Optional[str] = schema_utils.StringOptions(
+        list(get_encoder_classes(TEXT).keys()),
+        default="parallel_cnn",
+        description="Encoder to use for this text feature.",
+    )
+
+    # TODO(#1673): Need some more logic here for validating against input features
+    tied: Optional[str] = schema_utils.String(
```

Comment:
Will move to the schema generation logic since you are right, we don't really need to define this in the schema if it's consistent across all features.

Reply:
Done!

Reply:
So after making this change, it looks like a bunch of the tests are failing now because they want to set tied to `None` which is not of type string. Was looking for a way to do that, however it got me thinking that since we want to create this schema object for the SDK that basically uses the schema definitions for each input feature type in order to create input/output features via the SDK using dot notation, it may be better to actually have these constants defined in the schema classes themselves like `name`, `type` and `tied`. So that the user can modify all these attributes when creating these objects via the SDK object. WDYT, @justinxzhao?

Reply:
@connor-mccorm missing some context, but would defining some tertiary base/common class for fields like this and deriving it work, e.g. "`CommonFeatureOptions`"

Reply:
Ya that's probably the move actually. I had that before but ran into an inheritance issue that I resolved by just removing the class entirely which was dumb of me. Didn't have the foresight to see that we would probably need that at the time. Will add now.